crates/thetawave_interface/src/player.rs

```diff
@@ -1,8 +1,48 @@
-use bevy::prelude::*;
+use crate::{
+    character::{Character, CharacterType},
+    spawnable::ProjectileType,
+};
+use bevy_ecs::prelude::Component;
+use bevy_ecs::system::Resource;
+use bevy_math::Vec2;
+use bevy_time::{Timer, TimerMode};
 use serde::Deserialize;
-use thetawave_interface::spawnable::ProjectileType;
 
-use crate::player::Character;
+#[derive(Resource, Debug)]
+pub struct PlayersResource {
+    pub player_data: Vec<Option<PlayerData>>,
+}
+
+#[derive(Debug, Clone)]
+pub struct PlayerData {
+    pub character: CharacterType,
+    pub input: PlayerInput,
+}
+
+impl Default for PlayersResource {
+    fn default() -> Self {
+        PlayersResource {
+            player_data: vec![None, None, None, None],
+        }
+    }
+}
```

Comment:
`PlayersResource` doesnt make any sense to me. Why is it Vec<Option<_>>? Why is the default 4 nones?

Reply:
It's for Nones for up to for possible players in the game. When a player slot is filled it becomes Some(PlayerData) for that slot.